pkgs/applications/office/gnucash/default.nix

```diff
@@ -97,12 +98,29 @@ stdenv.mkDerivation rec {
   enableParallelChecking = true;
   checkTarget = "check";
 
+  passthru.docs = stdenv.mkDerivation {
+    pname = "gnucash-docs";
+    inherit (finalAttrs) version;
+
+    src = fetchFromGitHub {
+      owner = "Gnucash";
+      repo = "gnucash-docs";
+      rev = finalAttrs.version;
```

Comment:
`version` is not part of the interface of the drv; you can't change it without digging into the various `src`. Since `finalAttrs` has a significant perf overhead, please use `rec` or let bindings instead.

Reply:
Ah, just noticed that that's how you initially had it. @AndersonTorres please don't recommend finalAttrs unless there's a good reason.

Reply:
Can you provide a list of **objectively** good reasons?

Reply:
Can you provide a list of _objectively_ good reasons?

EDIT:
Let's do even better:
@drupol is way more interested in spreading the word of "`finalAttrs` over `rec`" than me.
Explain to they what are those good reasons.

Reply:
Hello colleagues!

I'm a bit away from reviewing and stuff because of a massive load of work and personal stuff. 

I usually advise the `finalAttr` pattern indeed but I'm not aware of any performance issue.

And even if it has some impact on perf, I would still recommend it since the `rec` implementation is more like a hack to me.

Anyway, I'll let you decide what to do in this PR which looks legit to me.

Cheers

Reply:
Hi @Atemu, it seems like others have addressed why they think `finalAttrs` makes sense here, and I tend to agree. Would you still be strongly opposed to the `finalAttrs` change?

Also minor correction, the first draft already had the `finalAttrs` change, but I misspelled it as `oldAttrs`. AndersonTorres's review was correcting the typo, not recommending `finalAttrs` (I had already done that).

Reply:
Before:

```
Benchmark 1: nix-instantiate -A gnucash
  Time (mean ± σ):     816.9 ms ±  38.8 ms    [User: 474.6 ms, System: 181.4 ms]
  Range (min … max):   781.2 ms … 885.0 ms    30 runs
```

After:

```
Benchmark 1: nix-instantiate -A gnucash
  Time (mean ± σ):     814.0 ms ±  41.6 ms    [User: 477.8 ms, System: 177.3 ms]
  Range (min … max):   773.8 ms … 893.2 ms    30 runs
```

It seems my data on the performance overhead was either way outdated or something was optimised in the mean time. IIRC it was a lot more significant than this (10-20%).

With the given data, there is no strong opposition from me but rather the question why you'd use it. There is not much point in being able to update the version in all places using overrideAttrs as most changes in version would need changes in various srcs aswell, so you might aswell override the src directly.

A packages' `version` attr is not really part of its "interface" (see https://discourse.nixos.org/t/preferred-method-for-defining-optional-features-in-packages/34565).

Reply:
OK, I don't have any strong opinions about using `finalAttrs`, so I removed it.